File: Dae/3월 18일/6064.py
import sys
input = sys.stdin.readline
T = int(input())
for i in range(T):
    M, N, x, y = map(int,input().split(' '))
    p = M*N
    count = -1
    if x <= M and y <= N:
        for j in range(x,p+1,M):
            if (j-1)%N + 1 == y:
                count = j
                break
            
    print(count)

File: Dae/3월 18일/6064.c
#include <stdio.h>

int main(void) {
    int M,N,x,y,T,count;
    scanf("%d",&T);
    for(int i=0;i<T;++i) {
        scanf("%d %d %d %d",&M,&N,&x,&y);
        count=0;
        int x1=1,y1=1;
        while (1) {
            if(x == x1 && y == y1) {
                printf("%d\n",count+1);
                break;
            }
            else if(x1 == M && y1 == N) {
                printf("-1\n");
                break;
            }
            if(x1 == M) {
                x1 = 1;
                y1++;
                count++;
            }
            if(x == x1 && y == y1) {
                printf("%d\n",count+1);
                break;
            }
            if(y1 == N) {
                y1 = 1;
                x1++;
                count++;
            }
            if(x == x1 && y == y1) {
                printf("%d\n",count+1);
                break;
            }
            x1++;
            y1++;
            count++;
        }
    }
    return 0;
}